src/System.Security.AccessControl/tests/Ace.Tests.cs

```diff
@@ -0,0 +1,482 @@
+            yield return CustomAce_CreateTestData(18, 1, 8, 0);
+            yield return CustomAce_CreateTestData(17, 1, 4, 0);
+        }
+
+        [Fact]
+        public void CustomAce_Constructor_Invalid()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new CustomAce((AceType)16, (AceFlags)15, new byte[1]));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new CustomAce((AceType)19, (AceFlags)1, new byte[1]));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new CustomAce((AceType)0, (AceFlags)2, new byte[4]));
+
+            foreach (AceType type in Enum.GetValues(typeof(AceType)))
+            {
+                Assert.Throws<ArgumentOutOfRangeException>(() => new CustomAce(type, (AceFlags)1, new byte[4]));
+            }
+        }
+
+        [Fact]
+        public void CustomAce_CreateBinaryForm_Invalid()
+        {
+            GenericAce ace = new CustomAce((AceType)19, (AceFlags)0, new byte[4]);
+            Assert.Throws<ArgumentNullException>("binaryForm", () => CustomAce.CreateFromBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => CustomAce.CreateFromBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentException>("binaryForm", () => CustomAce.CreateFromBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentException>("binaryForm", () => CustomAce.CreateFromBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Fact]
+        public void CustomAce_GetBinaryForm_Invalid()
+        {
+            GenericAce ace = new CustomAce((AceType)19, (AceFlags)0, new byte[4]);
+            Assert.Throws<ArgumentNullException>("binaryForm", () => ace.GetBinaryForm(null, 1));
+            Assert.Throws<ArgumentOutOfRangeException>("offset", () => ace.GetBinaryForm(new byte[1], -1));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength + 1], 2));
+            Assert.Throws<ArgumentOutOfRangeException>("binaryForm", () => ace.GetBinaryForm(new byte[ace.BinaryLength], 1));
+        }
+
+        [Theory]
+        [MemberData(nameof(CustomAce_TestObjects))]
+        public void CustomAce_GetBinaryForm(GenericAce testAce, byte[] expectedBinaryForm, int testOffset)
+        {
+            byte[] resultBinaryForm = new byte[testAce.BinaryLength + testOffset];
+            testAce.GetBinaryForm(resultBinaryForm, testOffset);
+            GenericAce_VerifyBinaryForms(expectedBinaryForm, resultBinaryForm, testOffset);
+        }
+
+        [Theory]
+        [MemberData(nameof(CustomAce_TestObjects))]
+        public void CustomAce_CreateFromBinaryForm(GenericAce expectedAce, byte[] testBinaryForm, int testOffset)
+        {
+            GenericAce resultAce = CustomAce.CreateFromBinaryForm(testBinaryForm, testOffset);
+            GenericAce_VerifyAces(expectedAce, resultAce);
+        }
+    }
+
+    public class CompoundAce_Tests : KnownAce_Tests
```

Comment:
I know that the product code has like 85 types in one file, but can we break these into multiple files?